src/hooks/useAxiosFunction.ts

```diff
@@ -1,44 +1,37 @@
-import { useEffect, useState, Dispatch, SetStateAction } from 'react';
-import { AxiosError, AxiosResponse } from 'axios';
-type setterFunctionArray = Dispatch<SetStateAction<any>>[];
+import { useEffect, useState } from 'react';
+import {
+  AxiosError,
+  AxiosResponse,
+  AxiosRequestConfig,
+  AxiosInstance,
+} from 'axios';
+import defaultAxiosInstance from '../apis/reviveme';
 
-interface Config {
-  axiosInstance: any;
-  method: string;
-  url: string;
-  requestConfig?: Record<string, any>;
-  setterFunctions?: setterFunctionArray;
-}
-
-const useAxiosFunction = () => {
+const useAxiosFunction = (
+  axiosInstance: AxiosInstance = defaultAxiosInstance,
+) => {
   const [response, setResponse] = useState<any[] | any>([]);
   const [error, setError] = useState<string>('');
   const [loading, setLoading] = useState<boolean>(true);
   const [controller, setController] = useState<AbortController>();
 
-  const axiosFetch = async (config: Config) => {
-    const {
-      axiosInstance,
-      method,
-      url,
-      requestConfig = {},
-      setterFunctions = [],
-    } = config;
+  const axiosFetch = async (
+    config: AxiosRequestConfig,
```

Comment:
NIT: add more specificity to what an AxiosRequestConfigs requires via a comment or some other manner